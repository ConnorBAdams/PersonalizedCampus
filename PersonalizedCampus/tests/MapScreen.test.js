/**
 * @jest-environment jsdom
 */

import React from 'react';
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import MapScreen from '../screens/mapScreen';
import * as firebase from 'firebase'
require('firebase/auth')
import { shallow, mount, render } from 'enzyme';
import sinon from 'sinon';

jest.mock('@react-navigation/native');

jest.mock('firebase', () => {
  return {
    initializeApp: jest.fn(() => {
      return {
        auth: jest.fn(() => {
          return {
            createUserWithEmailAndPassword: jest.fn((para1, para2) => {
              return new Promise(function(resolve, reject) {
                resolve({
                  email: 'test@test.com',
                  uid: '12345678abcdefg'
                });
                reject({ message: 'error!' });
              });
            }),
            signOut: jest.fn(() => {
              return new Promise(function(resolve, reject) {
                resolve('Success');
                reject({ message: 'error!' });
              });
            }),
            onAuthStateChanged: jest.fn(() => {
              return {
                email: 'test@test.com',
                uid: '12345678abcdefg'
              };
            }),
            signInWithEmailAndPassword: jest.fn((para1, para2) => {
              return new Promise(function(resolve, reject) {
                reject({ message: 'error!' });
              });
            }),
          };
        })
      };
    }),
    auth: jest.fn(() => {
      return {
        createUserWithEmailAndPassword: jest.fn((para1, para2) => {
          return new Promise(function(resolve, reject) {
            resolve({
              email: 'test@test.com',
              uid: '12345678abcdefg'
            });
            reject({ message: 'error!' });
          });
        }),
        signOut: jest.fn(() => {
          return new Promise(function(resolve, reject) {
            resolve('Success');
            reject({ message: 'error!' });
          });
        }),
        onAuthStateChanged: jest.fn(() => {
          return {
            email: 'test@test.com',
            uid: '12345678abcdefg'
          };
        }),
        signInWithEmailAndPassword: jest.fn((para1, para2) => {
          return new Promise(function(resolve, reject) {
            reject({ message: 'error!' });
          });
        }),
      };
    })
  };
});


it("Snapshot of Map Screen", () => {
  const wrapper = shallow(<MapScreen />);
  expect(wrapper).toMatchSnapshot();
});

it('Testing Location Permissions Error Message', () => {
    const context = { name: 'foo' };
    const wrapper = mount(<MapScreen />, { context });
    wrapper.setContext({status: 'granted'});
    expect(wrapper.contains('Permission to access location was denied, is is required for the map')).toBeFalsy();
  });