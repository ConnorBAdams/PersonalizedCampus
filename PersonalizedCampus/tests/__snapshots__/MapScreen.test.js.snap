// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Map Screen 1`] = `ShallowWrapper {}`;
